pkg/utils/yaml/yaml.go

```diff
@@ -25,3 +28,31 @@ func Serialize[T any](objs ...T) ([][]byte, error) {
 	}
 	return r, nil
 }
+
+// StripNull removes all null fields from the provided yaml.
+func StripNull(resources []byte) ([]byte, error) {
+	uList, err := unstructuredutil.YamlToUnstructured(resources)
```

Comment:
The underlying implementations of `Unstructured` expect to be working with Kubernetes objects. This package should contain YAML generic functionality only so I don't think its appropriate for here.

We can either remove the dependency on unstructured; I don't think that would be too difficult. Alternatively, move to a more appropriate location such as the `unstructured` util package.